### PR TITLE
Group Chat

### PR DESCRIPTION
**Pull request by Zep3k**
created the ui for the group chat
Screenshot_20240423-151946_SkySync
Screenshot_20240423-151950_SkySync
Screenshot_20240423-151954_SkySync

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/components/GroupChat.kt [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+// @Composable
+// @Preview
+// fun GroupChatPreview() {
+//    val image: ImageVector? = null
+//    val groups = listOf(Quadruple("Group 1", image, "Last message", "Last message time"),
+//        Quadruple("Group 2",image, "Last message", "Last message time"),
+//        Quadruple("Group 3",image,"Last message", "Last message time")
+//    )
+//    GroupChat(GroupsImageLastmsgLastmsgtime = groups, onClick = {}, paddingValues =
+// PaddingValues(0.dp))
```
[the remark this answers is not in the document]
Comment (Zep3k) [reply app/src/main/java/ch/epfl/skysync/components/GroupChat.kt:199]: this is just to preview the ui so i think that having it at 0 is better just to have it fill all the screen and to have a better view of it

File: app/src/main/java/ch/epfl/skysync/components/GroupChat.kt [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+ * Composable function to display a group chat UI.
+ *
+ * @param GroupsImageLastmsgLastmsgtime A list of quadruples representing group data containing
+ *   group name, image, last message, and last message time.
+ * @param onClick Callback triggered when a group is clicked.
+ * @param paddingValues Padding values for the column.
+ */
+@Composable
+fun GroupChat(
+    GroupsImageLastmsgLastmsgtime: List<Quadruple<String, ImageVector, String, String>>,
```
[the remark this answers is not in the document]
Comment (Zep3k) [reply app/src/main/java/ch/epfl/skysync/components/GroupChat.kt]: I agree

File: app/src/main/java/ch/epfl/skysync/components/GroupChat.kt [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import ch.epfl.skysync.ui.theme.lightOrange
+
+data class Quadruple<A, B, C, D>(val first: A, val second: B?, val third: C, val fourth: D)
```
[the remark this answers is not in the document]
Comment (Zep3k) [reply app/src/main/java/ch/epfl/skysync/components/GroupChat.kt]: indeed it's better idea

**Review by mathieumognol — APPROVED** [review]
LGTM

**Pull request merged**